crates/router/src/utils.rs

```diff
@@ -70,3 +76,52 @@ pub mod error_parser {
 pub fn generate_id(length: usize, prefix: &str) -> String {
     format!("{}_{}", prefix, nanoid!(length, &consts::ALPHABETS))
 }
+
+pub trait ConnectorResponseExt: Sized {
+    fn get_response(self) -> RouterResult<types::Response>;
+    fn get_error_response(self) -> RouterResult<types::Response>;
+    fn get_response_inner<T: DeserializeOwned>(self, type_name: &str) -> RouterResult<T> {
+        self.get_response()?
+            .response
+            .parse_struct(type_name)
+            .change_context(errors::ApiErrorResponse::InternalServerError)
+    }
+}
+
+impl<E> ConnectorResponseExt
+    for Result<Result<types::Response, types::Response>, error_stack::Report<E>>
+{
+    fn get_error_response(self) -> RouterResult<types::Response> {
+        self.change_context(errors::ApiErrorResponse::InternalServerError)
+            .attach_printable("Error while receiving response")
+            .and_then(|inner| match inner {
+                Ok(res) => {
+                    logger::error!(response=?res);
+                    Err(errors::ApiErrorResponse::InternalServerError)
+                        .into_report()
+                        .attach_printable(format!(
+                            "Expecting error response, received response: {res:?}"
+                        ))
+                }
+                Err(err_res) => Ok(err_res),
+            })
+            .change_context(errors::ApiErrorResponse::InternalServerError)
```

Comment:
its already return `InternalServerError` why again change context is required and below function too.